src/env-inl.h

```diff
@@ -623,11 +615,13 @@ inline void Environment::set_can_call_into_js(bool can_call_into_js) {
 }
 
 inline bool Environment::has_run_bootstrapping_code() const {
-  return has_run_bootstrapping_code_;
+  return principal_realm_->has_run_bootstrapping_code();
 }
 
 inline void Environment::DoneBootstrapping() {
-  has_run_bootstrapping_code_ = true;
+  CHECK(has_run_bootstrapping_code());
+  // TODO(legendecas): distinguish base objects with realms.
```

Comment:
Did you mean associating BaseObjects with the realms where they are created, instead of environments? (it seems to be a comment that belongs to the BaseObject class definition?)

Reply:
I mean the `base_object_created_by_bootstrap_` tracked below this line. The count should be tracked for each realm, when the base object is disentangled with the Environment.

Reply:
```suggestion
  // TODO(legendecas): track base objects by realms instead of environments.
```